cli/tests/activate/nested-activate-fish.exp

```diff
@@ -0,0 +1,40 @@
+# Activate a project environment using --dir
+
+set dir [lindex $argv 0]
+set flox $env(FLOX_BIN)
+set timeout 10
+
+spawn $flox activate --dir $dir/outer
+expect_after {
+  timeout { exit 1 }
+  eof { exit 2 }
+  "*\n" { exp_continue }
+  "*\r" { exp_continue }
+}
+expect "You are now using the environment"
+
+send "$flox activate --dir $dir/inner\r"
+expect "You are now using the environment"
+
+send "inner\r"
+expect "the inner alias is defined"
+
+send "outer\r"
+expect "the outer alias is defined"
+
+send "current\r"
+expect "the current alias is inner"
+
```

Comment:
suggestion followup: I think we can make all the assertions from bats, which tends to cause us less problems than asserting in expect
```suggestion
```

Reply:
You need to have _something_ to expect after each of the commands, and the only other thing you have to look for is the contents of the prompt that will vary by shell type. So these alias outputs are the logical thing to look for here, and note that we make the same assertions _again_ from bats as well.

Reply:
You could `send "inner && outer && current && exit"`. I'm just averse to `expect`s because they've flaked in the past

Reply:
Having used `expect` for 30+ years I'll say it's got its quirks but is battle-tested and reliable once an expect stream has been established, which at this point in the script it will have. I'm always a fan of "belt & braces" so asserting the expected output in both places can only be helpful, but sure we can change it to blindly fire off the commands and exit.

Reply:
commit fd738c8369f083a62380707ab252a3d529d7ecc5